tests/scripts/run-xunit-slice.py

```diff
@@ -0,0 +1,82 @@
+import json
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+##########################################################################
+# Argument Parser
+##########################################################################
+
+description = 'Tool to generate slice<number>.txt files based off of a slice json file (see perf-slices.json for format)'
+
+parser = argparse.ArgumentParser(description=description)
+
+parser.add_argument('-slice', dest='slice_number', default=-1, type=int)
+parser.add_argument('-slice-file', dest='slice_file', default=None)
+parser.add_argument('-slice-dir', dest='slice_dir', default=None)
+
+##########################################################################
+# Helper Functions
+##########################################################################
+
+def validate_args(args):
+    """ Validate all of the arguments parsed.
+    Args:
+        args (argparser.ArgumentParser): Args parsed by the argument parser.
+    Returns:
+        (arch, build_type, clr_root, fx_root, fx_branch, fx_commit, env_script)
+            (str, str, str, str, str, str, str)
+    Notes:
+    If the arguments are valid then return them all in a tuple. If not, raise
+    an exception stating x argument is incorrect.
+    """
+
+    slice_number = args.slice_number
+    slice_file = args.slice_file
+    slice_dir = args.slice_dir
+
+    def validate_arg(arg, check):
```

Comment:
This function can be moved out of the validate_args function, and should be unless there is a pressing need to define it in this scope

Reply:
I copied this from other python scripts that we have developed over time in coreclr. This is basically just our standard practice. Also, validate_arg doesn't need to be used anywhere else, so keeping it in scope of validate_args seemed reasonable.